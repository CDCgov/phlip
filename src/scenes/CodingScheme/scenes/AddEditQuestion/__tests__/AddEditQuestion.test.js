import React from 'react'
import { shallow, mount } from 'enzyme'
import { AddEditQuestion } from '../index'

const props = {
  actions: {
    updateQuestionRequest: () => { },
    addChildQuestionRequest: () => { },
    addQuestionRequest: () => { }
  },
  match: {
    url: '`/project/2/coding-scheme/add`'
  },
  formActions: {
    reset: () => { }
  },
  location: {
    state: {}
  },
  form: {}
}

describe('CodingScheme scene - AddEditQuestion Component', () => {
  test('should render correctly', () => {
    expect(shallow(<AddEditQuestion {...props} />)).toMatchSnapshot()
  })
})