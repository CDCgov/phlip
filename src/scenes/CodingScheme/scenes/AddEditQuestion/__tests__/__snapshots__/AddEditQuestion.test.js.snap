// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CodingScheme scene - AddEditQuestion Component should render correctly 1`] = `
<FormModal
  form="questionForm"
  handleSubmit={[Function]}
  initialValues={
    Object {
      "includeComment": false,
      "isCategoryQuestion": false,
      "possibleAnswers": Array [
        Object {},
        Object {},
        Object {},
      ],
      "questionType": 4,
    }
  }
  maxWidth="md"
  open={true}
  validate={[Function]}
>
  <FlexGrid
    align="stretch"
    circular={false}
    container={true}
    flex={false}
    justify="stretch"
    padding="20px 20px 0"
    raised={false}
    style={
      Object {
        "minWidth": 890,
      }
    }
    type="column"
  >
    <FlexGrid
      align="stretch"
      circular={false}
      container={true}
      flex={false}
      justify="stretch"
      padding={0}
      raised={false}
      style={Object {}}
      type="column"
    >
      <ModalTitle
        title="Add New Question"
      />
      <ModalContent>
        <FlexGrid
          align="stretch"
          circular={false}
          container={true}
          flex={false}
          justify="stretch"
          padding={0}
          raised={false}
          style={Object {}}
          type="row"
        >
          <FlexGrid
            align="stretch"
            circular={false}
            container={false}
            flex={true}
            justify="stretch"
            padding="0 10px 20px 0"
            raised={false}
            style={Object {}}
            type="column"
          >
            <Field
              component={[Function]}
              disabled={true}
              label="Question"
              multiline={true}
              name="text"
              placeholder="Enter question"
              required={true}
              shrinkLabel={true}
              smallLabel={true}
            />
          </FlexGrid>
          <FlexGrid
            align="stretch"
            circular={false}
            container={false}
            flex={false}
            justify="stretch"
            padding={0}
            raised={false}
            style={Object {}}
            type="column"
          >
            <Field
              component={[Function]}
              defaultValue={4}
              disabled={true}
              fullWidth={true}
              label="Answer Type"
              name="questionType"
              onChange={[Function]}
              options={
                Array [
                  Object {
                    "label": "Binary",
                    "value": 1,
                  },
                  Object {
                    "label": "Checkbox",
                    "value": 3,
                  },
                  Object {
                    "label": "Radio Button",
                    "value": 4,
                  },
                  Object {
                    "label": "Text Field",
                    "value": 5,
                  },
                  Object {
                    "label": "Tabbed",
                    "value": 2,
                  },
                ]
              }
              required={true}
            />
          </FlexGrid>
        </FlexGrid>
        <FlexGrid
          align="stretch"
          circular={false}
          container={true}
          flex={false}
          justify="stretch"
          padding={0}
          raised={false}
          style={Object {}}
          type="row"
        >
          <FlexGrid
            align="stretch"
            circular={false}
            container={false}
            flex={true}
            justify="stretch"
            padding="0 10px 20px 0"
            raised={false}
            style={Object {}}
            type="column"
          >
            <Field
              component={[Function]}
              disabled={true}
              label="Coding directions"
              name="hint"
              placeholder="Enter any special directions or considerations to display when coding this question"
              shrinkLabel={true}
              smallLabel={true}
            />
          </FlexGrid>
        </FlexGrid>
        <FieldArray
          answerType={4}
          component={[Function]}
          isEdit={false}
          name="possibleAnswers"
        />
        <FlexGrid
          align="stretch"
          circular={false}
          container={true}
          flex={false}
          justify="stretch"
          padding={0}
          raised={false}
          style={Object {}}
          type="row"
        >
          <FlexGrid
            align="stretch"
            circular={false}
            container={false}
            flex={true}
            justify="stretch"
            padding={0}
            raised={false}
            style={
              Object {
                "paddingLeft": "47px",
              }
            }
            type="column"
          >
            <Field
              component={[Function]}
              disabled={true}
              label="Include comment box"
              name="includeComment"
            />
          </FlexGrid>
        </FlexGrid>
      </ModalContent>
    </FlexGrid>
    <WithStyles(ModalActions)
      actions={
        Array [
          Object {
            "onClick": [Function],
            "otherProps": Object {
              "aria-label": "Cancel and close form",
            },
            "type": "button",
            "value": "Cancel",
          },
          Object {
            "disabled": true,
            "otherProps": Object {
              "aria-label": "Save form",
            },
            "type": "submit",
            "value": <React.Fragment>
              Add
            </React.Fragment>,
          },
        ]
      }
      style={
        Object {
          "margin": 0,
          "paddingBottom": 15,
          "paddingTop": 15,
        }
      }
    />
  </FlexGrid>
</FormModal>
`;
